arrow/src/test/java/org/apache/calcite/adapter/arrow/ArrowAdapterTest.java

```diff
@@ -518,6 +518,38 @@ static void initializeArrowState(@TempDir Path sharedTempDir) throws IOException
         .explainContains(plan);
   }
 
+  @Test void testDateProject() {
+    String sql = "select HIREDATE from EMP";
+    String plan = "PLAN=ArrowToEnumerableConverter\n"
+        + "  ArrowProject(HIREDATE=[$4])\n"
+        + "    ArrowTableScan(table=[[ARROW, EMP]], fields=[[0, 1, 2, 3, 4, 5, 6, 7]])\n\n";
+    String result = "HIREDATE=1970-01-01 00:00:04\n"
+        + "HIREDATE=1970-01-01 00:00:04\n"
```

Comment:
Is this result correct? The date type should not include hours, minutes, and seconds.

Reply:
It seems that calcite has a bug in arrow's handling of date type. I will open a new jira to fix it